### ISSUE TITLE
[UX] Timeline progress improvements, May edition

### ISSUE THREAD

**Issue by jonbell-lot23** [replayio/design#52]
**Background/the problem we're trying to solve**

We have made many improvements to our timeline over the last year, and this is the next step forward. These items should be as clear as possible when using our app:

1. Where the playhead currently is [good]
2. What regions are ready to play quickly [misleading]
3. Messaging when things are slow/laggy/not ready [bad]

**Details: What regions are ready to play quickly**

We currently use a slightly different color in the style of Netflix or YouTube to say "this region is fully loaded and ready to go. Click here and everything's gonna work great." But it doesn't. It is often slow.

Proposed fix: we need better messaging here.

**Details: Messaging when things are slow/laggy/not ready**

It's possible to click into a portion of the timeline and not get a meaningful explanation about what's going wrong.

Proposed fix: we also need better messaging here.

---

**Proposed plan**

1. First we should ship https://github.com/RecordReplay/devtools/issues/6671. Let's get that out as designed, then we can build on that foundation.
2. Once that's live, we need to think about how/where we're going to message stuff. Coming soon.

**The design that we'll probably land on**

* Mousing over the loading indicator on the bottom right will light up indexed regions in the timeline
* Mousing over the play button on the bottom left might do something -- stay tuned
* When we know things are slow, we'll put a spinner on the playhead indicator (aka clown nose)
* You should be able to click absolutely anywhere, loaded or not. Just a matter of messaging.
* Mousing over the loading clown nose might show a tooltip that shows progress.

**Other considerations**

There's more thinking to do around unloaded regions. (See tickets below) Stay tuned for that.

**Resources**

Ticket: [Difficulty using player when parts of recording are unloaded](https://github.com/RecordReplay/devtools/issues/6622)
Notion: [Josh, Brian, Jon notes](https://www.notion.so/replayio/Video-buffering-discussion-86dd30a4b30745ef86948fb683ba02f8)

**Comment by jcmorrow** [comment]
I want to add some not-very-fun but sadly important nuance here:

> We currently use a slightly different color in the style of Netflix or YouTube to say "this region is fully loaded and ready to go. Click here and everything's gonna work great." But it doesn't. It is often slow.

I don't think this is true? I think right now this is supposed to mean "we have loaded paints for this area so the video should work". Right now, AFAIK, there is no bar that tells the user "this part of the recording is loaded" (though we have discussed adding one when hovering of the new loading UI". If we do that, we should recognize that there will be at least 4 different ways for the timeline to look visually, with a different significance to each one:

1) unloaded
2) loaded/loading, video not ready
3) loaded/loading, video ready
4) loaded/loading, in the "past" (to the left of the current play position)

And I'm sure that could get confusing.

**Comment by jcmorrow** [comment]
And, it's worth noting, even when we *do* show the user which parts are loaded, just because the part of the recording where you are currently paused is loaded does not mean "everything is gonna work". Lots of stuff will still wait for indexing to be complete before it works.

**Comment by bvaughn** [comment]
I _think_ we talked about the ability to differentiate between index/loaded and _ready for playback_ (maybe connected to the paint regions?) – if we have that ability (I think we said we did?) then I think _that's_ what we should be showing with the timeline progress bar, because _that ui_ is primarily about playback.

**Comment by jcmorrow** [comment]
Yep, if you see the new 4th bar on the protocol timeline (the bottom blue one), that is what we know about where repaints fall in the replay. That doesn't mean we have the paint data just yet, we would still need to download it, but at least we could start doing that if the user clicked there.

**Comment by bvaughn** [comment]
Blah. _That's_ where it went. It used to be turned on via `app.features.advancedTimeline` (right?) but that stopped working for me. It's `app.features.protocolTimeline` now I guess. That being said, I only see 3 lines 😄

**Comment by jcmorrow** [comment]
Yeah I introduced a bug in the paints calculation this morning where we would try and make it `Infinity%` wide when it was all loaded 😅

**Comment by jcmorrow** [comment]
It should be fixed now

**Comment by jonbell-lot23** [comment]
I wrote a comment but revising it back because it didn't add a lot of new information

**Comment by jonbell-lot23** [comment]
This ticket is causing a bit of confusion in my workflow because it's not closed but I'm also waiting on some other focus mode stuff before I do anything else. Sending to next iteration.

**Comment by jonbell-lot23** [comment]
Hey @bvaughn and @jcmorrow, I've filed a new ticket for adding a loading indicator onto the playhead when things are not yet loaded: https://github.com/RecordReplay/devtools/issues/6833

It is likely that with that change I'll be able to close out this ticket, but stay tuned.

Update: I put together a design so if we agree with it we can roll it out. (I'll need to cut an icon, stay tuned for that)

**Comment by jonbell-lot23** [comment]
I am closing this ticket in favour of other ones. The best way to keep track is on this Discord page:
https://www.notion.so/replayio/Loading-buffering-loading-states-design-discussion-2c6b4e0a13f845ddb09221a3a3231631
